### PR TITLE
Emphasize that HttpClientFactory depends on M.E.DI

### LINKED ISSUE #13850: Point out this only serves users of the MS DI container or ASP .NET Core.
The only way to configure HttpClientFactory is through the MS DI container. If you are using any other DI container, that container is forced to support the MS DI container infrastructure or there is no way for you to use IHttpClientFactory yourself. 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 7b506236-b211-eb9d-b571-5efecdde76c8
* Version Independent ID: 24434b0e-0b82-e477-9440-2cc8f135df75
* Content: [Use HttpClientFactory to implement resilient HTTP requests](https://docs.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests)
* Content Source: [docs/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests.md](https://github.com/dotnet/docs/blob/master/docs/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests.md)
* Product: **dotnet**
* Technology: **dotnet-ebooks**
* GitHub Login: @nishanil
* Microsoft Alias: **nanil**

### PR DESCRIPTION
**Pull request by mvelosop**
## Summary

And a note to emphasize that `HttpClientFactory` is tied to `Microsoft.Extensions.DependencyInjection`

Fixes https://github.com/dotnet/docs/issues/13850

### CODE REVIEW

File: docs/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests.md [hunk 1]
```diff
@@ -30,6 +30,9 @@ To address those mentioned issues and make the management of `HttpClient` instan
 - `HttpClient` already has the concept of delegating handlers that could be linked together for outgoing HTTP requests. You register HTTP clients into the factory and you can use a Polly handler to use Polly policies for Retry, CircuitBreakers, and so on.
 - Manage the lifetime of `HttpClientMessageHandlers` to avoid the mentioned problems/issues that can occur when managing `HttpClient` lifetimes yourself.
 
+> [!NOTE]
+> `HttpClientFactory` is tightly tied to the Microsoft.Extensions.DependencyInjection implementation. You might want to read this [GitHub discussion]((https://github.com/aspnet/Extensions/issues/1345)) about using other dependency injection containers.
```
Comment (mairaw) [inline docs/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests.md]:
```suggestion
> `HttpClientFactory` is tightly tied to the `Microsoft.Extensions.DependencyInjection` implementation. For more information about using other dependency injection containers, see this [GitHub discussion](https://github.com/aspnet/Extensions/issues/1345).
```

Comment (mairaw) [reply docs/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests.md]: clarify what Microsoft.Extensions.DependencyInjection is perhaps? It's a NuGet package, right? And how is it tied to that? Is it part of that package or what?

File: docs/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests.md [hunk 2]
```diff
@@ -174,14 +177,17 @@ Up to this point, the code shown is just performing regular Http requests, but t
 
 ## Additional resources
 
-- **Using HttpClientFactory in .NET Core** \
+- **Using HttpClientFactory in .NET Core**  
   [https://docs.microsoft.com/aspnet/core/fundamentals/http-requests](/aspnet/core/fundamentals/http-requests)
 
-- **HttpClientFactory GitHub repo** \
+- **HttpClientFactory GitHub repo**
```
Comment (mairaw) [inline docs/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests.md]:
```suggestion
- **HttpClientFactory GitHub repository**  
```

Comment (mairaw) [reply docs/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests.md]: Technically the repo is aspnet/Extensions. So maybe say HttpClientFactory source code (GitHub)?

**Review by mairaw — CHANGES_REQUESTED** [review]
Thanks @mvelosop. Left some suggestions.

File: docs/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests.md (hunk 1 above)
Comment (mvelosop) [reply docs/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests.md]: Yep, at this level it's more conceptual than package specific, will clarify it.

File: docs/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests.md [hunk 3]
```diff
@@ -174,14 +177,17 @@ Up to this point, the code shown is just performing regular Http requests, but t
 
 ## Additional resources
 
-- **Using HttpClientFactory in .NET Core** \
+- **Using HttpClientFactory in .NET Core**  
   [https://docs.microsoft.com/aspnet/core/fundamentals/http-requests](/aspnet/core/fundamentals/http-requests)
 
-- **HttpClientFactory GitHub repo** \
+- **HttpClientFactory source code in the .NET Extensions GitHub repository**
```
Comment (mairaw) [inline docs/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests.md]:
```suggestion
- **HttpClientFactory source code in the `aspnet/Extensions` GitHub repository**  
```

File: docs/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests.md [hunk 4]
```diff
@@ -174,14 +177,17 @@ Up to this point, the code shown is just performing regular Http requests, but t
 
 ## Additional resources
 
-- **Using HttpClientFactory in .NET Core** \
```
Comment (mairaw) [inline docs/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests.md]: loc confirmed they fixed the issue on the MT engine so we can use those again - just FYI

**Review by mairaw — APPROVED** [review]

**Comment by mairaw** [comment]
Feel free to squash and merge when you're ready

File: docs/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests.md (hunk 4 above)
Comment (mvelosop) [reply docs/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests.md]: Oh, great 😊

The trailing two blank spaces are very easy to miss!

Had been removing them 🙁, so back to adding them!

**Pull request merged**